base/array.jl

```diff
@@ -656,6 +656,20 @@ function collect_to_with_first!(dest, v1, itr, st)
     return grow_to!(dest, itr, st)
 end
 
+function setindex_widen!(dest::Array{T}, el, i) where T
```

Comment:
I suppose this could already be fully generic for `AbstractArray`?

Reply:
Wouldn't that be nice

Reply:
Ok should work for AbstractArrays now